apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_permission.exs

```diff
@@ -0,0 +1,16 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_permissions, primary_key: false) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource, references(:resources, on_delete: :nothing, column: :name, type: :string)
+      add :permission, references(:permissions, on_delete: :nothing, column: :name, type: :string)
+
+      timestamps
+    end
+    create index(:roles_permissions, [:role_id])
```

Comment:
I believe it will. The index is specifying an order to store the keys. So you are telling it to build a lookup index ordered by `role_id`, then `resource, then `permission`. If you go to look up records by `role_id` it should be able to utilize the index to retrieve everything tied to it.

Of course the best way to test this is to generate a bunch of data and analyze the query plan. I believe in PostgreSQL this is done with the [EXPLAIN](http://www.postgresql.org/docs/9.5/static/using-explain.html) keyword.

The issue with small data sets is that the database may choose to use a table scan over the index seek if the table is small enough. An index seek requires locating the rows of interest, gathering the keys, then fetching the data. A table scan would be like loading all data then filtering it out afterwards. As it grows the statistics should eventually steer the database to use the index. 

Again this is why it's best to not worry about indexes unless there are some obvious ones or you have done some performance analysis.